backend/oasst_backend/user_repository.py

```diff
@@ -1,16 +1,32 @@
 from typing import Optional
+from uuid import UUID
 
 from oasst_backend.models import ApiClient, Message, User
+from oasst_shared.exceptions import OasstError, OasstErrorCode
 from oasst_shared.schemas import protocol as protocol_schema
 from oasst_shared.schemas.protocol import LeaderboardStats
 from sqlmodel import Session, func
+from starlette.status import HTTP_403_FORBIDDEN
 
 
 class UserRepository:
     def __init__(self, db: Session, api_client: ApiClient):
         self.db = db
         self.api_client = api_client
 
+    def query_frontend_user(self, auth_method: str, username: str, api_client_id: UUID) -> Optional[User]:
+        if not self.api_client.trusted and api_client_id != self.api_client.id:
+            # Unprivileged API client asks for foreign user
+            raise OasstError("Forbidden", OasstErrorCode.API_CLIENT_NOT_AUTHORIZED, HTTP_403_FORBIDDEN)
+
+        user: User = (
+            self.db.query(User)
+            .filter(User.auth_method == auth_method, User.username == username, User.api_client_id == api_client_id)
+            .first()
+        )
+
+        return user
```

Comment:
General question @yk: What is the correct return value here we return `null`, empty response, or 404 error when the requested user is not found in the db?

Reply:
Have made it 404 in this scenario for now but let me know if this needs changes

Reply:
Getting 404 response seems reasonable to me.

Reply:
404 sounds good